spec/requests/api/v2/providers/courses/publishable_spec.rb

```diff
@@ -1,31 +1,27 @@
 require "rails_helper"
 
 describe "Publishable API v2", type: :request do
-  let(:user)         { create(:user) }
+  let(:user) { TestSetup.user }
   let(:organisation) { create(:organisation, users: [user]) }
-  let(:provider)     { create :provider, organisations: [organisation] }
-  let(:payload)      { { email: user.email } }
-  let(:token)        { build_jwt :apiv2, payload: payload }
-  let(:credentials) do
-    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  let(:provider) { create :provider, organisations: [organisation] }
+  let(:payload) { { email: user.email } }
+  let(:token) { build_jwt :apiv2, payload: payload }
+  let(:credentials) { ActionController::HttpAuthentication::Token.encode_credentials(token) }
+  let(:setup) do
+    TestSetup.new(with_courses: {
+      course: { provider: provider },
```

Comment:
I think the goal should be to avoid having to do this. I think we should change the factories to work with `find_or_create :provider`, which would mean we don't have to provide a provider when creating courses, etc.